### PR TITLE
Clarify array reference counting

### PR DESCRIPTION
**Pull request by r-barnes**

### CODE REVIEW

File: include/af/array.h [hunk 1]
```diff
@@ -203,7 +203,9 @@ namespace af
 #endif
 #endif
         /**
-            Creates an array from an \ref af_array handle
+            Creates an array from an \ref af_array handle. Does not increment 
+            a reference counter: the array assumes it has ownership of the
```
Comment (9prady9) [inline include/af/array.h]: Nit: the array assumes ownership of the handle.

File: include/af/array.h [hunk 2]
```diff
@@ -1573,7 +1574,9 @@ extern "C" {
     AFAPI af_err af_release_array(af_array arr);
 
     /**
-       Increments an \ref af_array reference count
+       Increments an \ref af_array reference count. The `out` pointer
+       need not be retained in situations where the reference count
+       is the only effect needed.
```
Comment (9prady9) [inline include/af/array.h]: I am confused here, what is out pointer retain ?

Comment (r-barnes) [reply include/af/array.h]: The function has two side effects. One is to set the value of the `out` pointer, which is the function's first argument. The other is to increment the `af_array`'s reference count. In situations where the only desired effect is the reference count, the `out` pointer can be tossed without further side-effects occurring.

File: include/af/array.h (hunk 1 above)
Comment (r-barnes) [reply include/af/array.h]: Can fix.

**Comment by r-barnes** [comment]
I'm not sure why `ci/win-opencl-k5000` is failing to build since these changes affect only comments.

**Comment by umar456** [comment]
Thats a known bug in clBlast. We can ignore that failure for this PR.

File: include/af/array.h (hunk 2 above)
Comment (r-barnes) [reply include/af/array.h]: I don't think this was strictly true, so I've removed this text.

File: include/af/array.h (hunk 1 above)
Comment (r-barnes) [reply include/af/array.h]: Fixed.

**Pull request merged**